config/v1/types_infrastructure.go

```diff
@@ -705,7 +705,30 @@ type AlibabaCloudResourceTag struct {
 
 // NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type NutanixPlatformSpec struct{}
+type NutanixPlatformSpec struct {
+	// prismCentral holds the endpoint and port to access the Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	PrismCentral NutanixPrismEndpoint `json:"prismCentral"`
+
+	// prismElements holds an array of endpoint and port data to access the Nutanix
+	// Prism Elements (clusters) of the Nutanix Prism Central
+	// +kubebuilder:validation:Required
+	PrismElements []NutanixPrismEndpoint `json:"prismElements"`
```

Comment:
How many of these are users expected to supply?

What does each different one represent?

Do they need to be named or have something unique within them?

How will consumers of this field know which of the elements to use?

Reply:
@JoelSpeed Prism Element (PE) is a single HCI cluster of physical nodes that's managed by a single entity (effectively, API endpoint). These Prism Element instances can then be aggregated and managed by a single Prism Central (PC) instance. We primarily target the PC instance for deployment, but there are some components that cannot (yet) be managed by PC (e.g. CSI storage), so we must also take PE input. For phase one, we are supporting only a single PE cluster for deployment, but in the future, it is possible that a single OCP cluster could span multiple PEs. Also, note that, by default, we ask the user for PE cluster name only, but in theory, it could also have a separate user/pass as well:

  ? Prism Central pc01.ocp.nutanix.com
  ? Prism Central Username admin@ocp.nutanix.com
  ? Prism Central Password [? for help] **********
  ? Cluster (Prism Element) nx01.ocp.nutanix.com

Reply:
Added the required name field (to identify it) in the Prism Element Endpoint structure. For this release, we only support one Prism Element (cluster) for an openshift cluster, where all the Nutanix resources (VMs, subnet, volume, etc.) used in the Openshift cluster locate. In the future, we may support the Nutanix resources (VMs, etc.) used in the Openshift cluster can come from multiple Prism Elements (clusters) of the Prism Cental.